### ISSUE TITLE
Angular 9.rc7: IVY ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies

### ISSUE THREAD

**Issue by tobiashoeft** [fullcalendar/fullcalendar-angular#270]
ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
 - @fullcalendar/core/types/input-types
 - @fullcalendar/core/datelib/env
 - @fullcalendar/core/datelib/duration
 - @fullcalendar/core/datelib/formatting
 - @fullcalendar/core/datelib/date-range
 - @fullcalendar/core/datelib/locale
 - @fullcalendar/core/validation
 - @fullcalendar/core/structs/event-source

Angular-CLI version 

  "@angular/cli": "9.0.0-rc.6",

.package.json 

 "dependencies": {
    "@angular/animations": "9.0.0-rc.7",
    "@angular/cdk": "~8.2.0",
    "@angular/common": "~9.0.0-rc.7",
    "@angular/compiler": "~9.0.0-rc.7",
    "@angular/core": "~9.0.0-rc.7",
    "@angular/forms": "~9.0.0-rc.7",
    "@angular/localize": "^9.0.0-rc.7",
    "@angular/material": "~8.2.0",
    "@angular/platform-browser": "~9.0.0-rc.7",
    "@angular/platform-browser-dynamic": "~9.0.0-rc.7",
    "@angular/router": "~9.0.0-rc.7",
    "@azure/ng-deploy": "^0.2.3",
    "@fullcalendar/angular": "^4.3.1",
    "@fullcalendar/core": "^4.3.1",
    "@fullcalendar/daygrid": "^4.3.0",
    "@fullcalendar/interaction": "^4.3.0",
    "@fullcalendar/list": "^4.3.0",
    "@fullcalendar/timegrid": "^4.3.0",

Reproduce:

$ ng serve

chunk {main} main.js, main.js.map (main) 2.07 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 715 bytes [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.15 kB [entry] [rendered]
chunk {scripts} scripts.js, scripts.js.map (scripts) 20.3 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 9.64 MB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 342 kB [initial] [rendered]
Date: 2020-01-11T08:15:53.980Z - Hash: ab65ca86f43f9aa9525a - Time: 13527ms

ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
 - @fullcalendar/core/types/input-types
 - @fullcalendar/core/datelib/env
 - @fullcalendar/core/datelib/duration
 - @fullcalendar/core/datelib/formatting
 - @fullcalendar/core/datelib/date-range
 - @fullcalendar/core/datelib/locale
 - @fullcalendar/core/validation
 - @fullcalendar/core/structs/event-source

** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

**Comment by josemadev** [comment]
FullCalendar seamlessly integrates with the Angular 7. It provides a component that exactly matches the functionality of FullCalendar’s standard API. I try upgrade it in this this pull request https://github.com/fullcalendar/fullcalendar-angular/pull/269. Try it, it works fine for me in angular 8.

**Comment by micobarac** [comment]
How do I add this pull request as a dependency through yarn?

**Comment by josemadev** [comment]
@micobarac yarn add https://github.com/josemadev/fullcalendar-angular#c12c0753f0ce354f29ea8db1a3cc2a871bca380e

**Comment by irustm** [comment]
fixes in #263

**Comment by tobiashoeft** [comment]
Thx u! 

Is the fix already on npm?

**Comment by micobarac** [comment]
ERROR in src/app/app.module.ts:4:36 - error TS2307: Cannot find module '@fullcalendar/angular'.

    4 import { FullCalendarModule } from '@fullcalendar/angular';

**Comment by josemadev** [comment]
Sorry i compiled the library in local extract it and you could do  yarn add <your path>/fullcalendar

[fullcalendar.zip](https://github.com/fullcalendar/fullcalendar-angular/files/4102327/fullcalendar.zip)

**Comment by micobarac** [comment]
Nope, I cannot install it like that:

```
~/angular9 (master)$ yarn add /fullcalendar
yarn add v1.21.1
[1/4] 🔍  Resolving packages...
error Package "/fullcalendar" refers to a non-existing file '"/fullcalendar"'.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**Comment by micobarac** [comment]
But this works:

`yarn add file:fullcalendar`

**Comment by josemadev** [comment]
Please @micobarac install @fullcalendar/core: yarn add @fullcalendar/core.
I have a 4.3.1 version

**Comment by micobarac** [comment]
@josemadev here is my `package.json`:

```json
{
  "name": "angular9",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve -o",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~9.0.0-rc.11",
    "@angular/common": "~9.0.0-rc.11",
    "@angular/compiler": "~9.0.0-rc.11",
    "@angular/core": "~9.0.0-rc.11",
    "@angular/forms": "~9.0.0-rc.11",
    "@angular/platform-browser": "~9.0.0-rc.11",
    "@angular/platform-browser-dynamic": "~9.0.0-rc.11",
    "@angular/router": "~9.0.0-rc.11",
    "@fullcalendar/angular": "file:fullcalendar",
    "@fullcalendar/core": "^4.3.1",
    "@fullcalendar/daygrid": "^4.3.0",
    "@fullcalendar/interaction": "^4.3.0",
    "@fullcalendar/list": "^4.3.0",
    "@fullcalendar/resource-timeline": "^4.3.0",
    "@fullcalendar/timegrid": "^4.3.0",
    "@fullcalendar/timeline": "^4.3.0",
    "@ngneat/until-destroy": "^7.1.0",
    "date-fns": "^2.9.0",
    "notyf": "^3.1.0",
    "rxjs": "~6.5.4",
    "tslib": "^1.10.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.900.0-rc.10",
    "@angular/cli": "~9.0.0-rc.10",
    "@angular/compiler-cli": "~9.0.0-rc.11",
    "@angular/language-service": "~9.0.0-rc.11",
    "@types/jasmine": "~3.5.1",
    "@types/jasminewd2": "~2.0.8",
    "@types/node": "^13.5.0",
    "codelyzer": "^5.2.1",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.4.1",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~2.1.1",
    "karma-jasmine": "~3.1.0",
    "karma-jasmine-html-reporter": "^1.5.1",
    "protractor": "~5.4.2",
    "ts-node": "~8.6.2",
    "tslint": "~5.20.1",
    "typescript": "~3.6.4"
  }
}

```

**Comment by luthor77** [comment]
I created an issue on angular about this problem (I didn't saw that already was here, and started in rc.7).

https://github.com/angular/angular/issues/35054

**Comment by JulienBourgain** [comment]
when will this fix be released ?

**Comment by codewoot** [comment]
Do we have a fix for this issue? It looks like a bug with @fullcalendar/angular as these dependencies are actually missing from the package.

**Comment by filipbech** [comment]
keeping my eye on this... v9rc14 seems to be the final version.. anything on this?

**Comment by irustm** [comment]
waiting this PR: #275

**Comment by micobarac** [comment]
Angular 9 full version has been officially released yesterday. Keep in mind this news.

**Comment by micobarac** [comment]
I compiled a branch angular9 lib via 'yarn build:lib' and added it to a project via `yarn add`, but I am getting the same error from the start of the issue:

```
ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
 - @fullcalendar/core/types/input-types
 - @fullcalendar/core/datelib/env
 - @fullcalendar/core/datelib/duration
 - @fullcalendar/core/datelib/formatting
 - @fullcalendar/core/datelib/date-range
 - @fullcalendar/core/datelib/locale
 - @fullcalendar/core/validation
 - @fullcalendar/core/structs/event-source
```

**Comment by domenichelfenstein** [comment]
> I compiled the `angular9` branch lib via `yarn build:lib` and added it to a project via `yarn add file:fullcalendar`, but I am getting the same error from the start of the issue:
> 
> ```
> ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
>  - @fullcalendar/core/types/input-types
>  - @fullcalendar/core/datelib/env
>  - @fullcalendar/core/datelib/duration
>  - @fullcalendar/core/datelib/formatting
>  - @fullcalendar/core/datelib/date-range
>  - @fullcalendar/core/datelib/locale
>  - @fullcalendar/core/validation
>  - @fullcalendar/core/structs/event-source
> ```

same here

**Comment by micobarac** [comment]
It only works if I update package.json dependencies to the latest versions.

**Comment by spelbreker** [comment]
> 
> 
> It only works if I update package.json dependencies to the latest versions.

can u show your package.json

**Comment by micobarac** [comment]
Install:

`yarn global add npm-check-updates`

Then run:

`ncu -u` to update package.json

Then run:

`yarn install`

**Comment by RicardoFazzi** [comment]
@micobarac I did that and I'm still getting the same error 

`ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
 - @fullcalendar/core/types/input-types
 - @fullcalendar/core/datelib/env
 - @fullcalendar/core/datelib/duration
 - @fullcalendar/core/datelib/formatting
 - @fullcalendar/core/datelib/date-range
 - @fullcalendar/core/datelib/locale
 - @fullcalendar/core/validation
 - @fullcalendar/core/structs/event-source`

**Comment by micobarac** [comment]
@RicardoFazzi you do the steps above in the root project of fullcalendar library, which updates dependencies to work with Angular 9 and Ivy. As a result, you will get fullcalendar folder inside dist folder, which you copy to your project's root and import it with yarn add file:fullcalendar. You could try to delete node_modules of your project and re-run yarn install. It worked for me.

**Comment by irustm** [comment]
@micobarac @RicardoFazzi  Look at this PR  #275, please. I cleaned up the typings. I think it is solve your problem

**Comment by micobarac** [comment]
Well, I must say that I am lost with all these PRs :(. If you have tested the changes and proved they work with Angular 9, please publish it to NPM, at least as a beta version.

**Comment by gfaganli** [comment]
Agree with @micobarac , waiting...

**Comment by TejasCMehta** [comment]
> @micobarac @RicardoFazzi Look at this PR #275, please. I cleaned up the typings. I think it is solve your problem

@irustm @micobarac @RicardoFazzi  can we have this published today?

**Comment by RicardoFazzi** [comment]
@micobarac I did everything you said, event reviewed and downloaded the PR 275, installed locally, and it fixed the original problem, but keeps telling me `Class FullCalendarModule is not an Angular module ` is like I'm not importing it from the right place or something? can anyone help me with this?
I'm doing:

```
**import { FullCalendarModule } from '@fullcalendar/angular';**
@NgModule({
  imports: [
    CommonModule,
    **FullCalendarModule**,
    AngularMaterialModule
  ],
  declarations: [GEFullCalendarComponent],
  exports: [GEFullCalendarComponent]
})

export class GEFullCalendarModule {

}
```

Thanks in advanced!

**Comment by butterfly-valley** [comment]
Using Angular 9.0.0 gives the same error:

> [ng] ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
[ng]  - @fullcalendar/core/types/input-types
[ng]  - @fullcalendar/core/datelib/env
[ng]  - @fullcalendar/core/datelib/duration
[ng]  - @fullcalendar/core/datelib/formatting
[ng]  - @fullcalendar/core/datelib/date-range
[ng]  - @fullcalendar/core/datelib/locale
[ng]  - @fullcalendar/core/validation
[ng]  - @fullcalendar/core/structs/event-source

**Comment by TejasCMehta** [comment]
@irustm @arshaw  when can we have beta release, please help us here, in my company we have projects to upgrade and this dependency is majorly holding us back to upgrade due to build errors. 

Please please please help!

**Comment by arshaw** [comment]
the `@fullcalendar/angular@4.4.5-beta` package has been published with all the latest changes from the angular9 branch. please let me and @irustm know if things still don't work

**Comment by IgDem** [comment]
@irustm @arshaw `@fullcalendar/angular@4.4.5-beta` works well - compilation with Angular 9 went with no problems. **Thank you!**

**Comment by EinarValholl** [comment]
`
 "@angular/animations": "~9.0.0",
    "@angular/cdk": "~9.0.0",
    "@angular/common": "~9.0.0",
    "@angular/compiler": "~9.0.0",
    "@angular/core": "~9.0.0",
    "@angular/elements": "~9.0.0",
    "@angular/forms": "~9.0.0",
    "@angular/localize": "^9.0.0",
    "@angular/material": "~9.0.0",
    "@angular/platform-browser": "~9.0.0",
    "@angular/platform-browser-dynamic": "~9.0.0",
    "@angular/router": "~9.0.0",
    "@fortawesome/angular-fontawesome": "^0.3.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.15",
    "@fortawesome/free-brands-svg-icons": "^5.7.1",
    "@fortawesome/free-regular-svg-icons": "^5.7.2",
    "@fortawesome/free-solid-svg-icons": "^5.7.2",
    "@fullcalendar/angular": "4.4.5-beta",
    "@fullcalendar/core": "^4.3.0",
    "@fullcalendar/interaction": "^4.3.0",
    "@fullcalendar/resource-timeline": "^4.3.0",
    "@ng-bootstrap/ng-bootstrap": "4.2.1",
`
![image](https://user-images.githubusercontent.com/5411602/74562126-b6f83b00-4f2f-11ea-9d56-a1a9cd73bd95.png)
With listed libraries above I've got an error on calendar click (dateClick emitter). Also, it's not possible to interact with events through `@fullcalendar/interaction` plugin (drag&drop functionality)

**Comment by Shelmy-pp** [comment]
i am using primeng 7 template and After upgrading to angular 9, I received an error saying that ERROR in Host Resource Loader: loader(E:/Angular9/myapp/src/assets/css/theme-indigo.css) returned a Promise It was working when I was using Angular 7 Angular 8, and I received it only after upgrading to 9.
Then Updated primeng,
"primeng": "^9.0.0-rc.4", and added "@fullcalendar/angular": "^4.4.5-beta",
But the error still exists
ERROR in HostResourceLoader: loader(E:/Angular9/myapp/src/assets/css/theme-indigo.css) returned a Promise

**Comment by chiragatlas** [comment]
when can we accept stable release of fullcalander that support angular 9? 
I can not upgrade my app to latest version as fullcalander is not supported yet.

**Comment by irustm** [comment]
@chiragatlas try use @fullcalendar/angular": "^4.4.5-beta

**Comment by chiragatlas** [comment]
@irustm yes I have use that, but in that drag and drop is not working.

**Comment by ricardofct** [comment]
@chiragatlas try to update @fullcalendar/core @fullcalendar/interaction @fullcalendar/resource-timeline to 4.4.0

**Comment by gitalvininfo** [comment]
Having the same issue after upgrading form Angular 8 to 9.

ERROR in The target entry-point "@fullcalendar/angular" has missing dependencies:
 - @fullcalendar/core/types/input-types
 - @fullcalendar/core/datelib/env
 - @fullcalendar/core/datelib/duration
 - @fullcalendar/core/datelib/formatting
 - @fullcalendar/core/datelib/date-range
 - @fullcalendar/core/datelib/locale
 - @fullcalendar/core/validation
 - @fullcalendar/core/structs/event-source

Any fixes or updates?

**Comment by netinc** [comment]
@gitalvininfo  I had this same issue yesterday migrating from 8 to 9.

I used `npm install @fullcalendar/angular@4.4.5-beta` then make sure the other @fullcalendar packages are at 4.4.0. This fixed it for me.

**Comment by gitalvininfo** [comment]
so what works for you is that you just install @fullcalendar/angular@4.4.5-beta ?

Thanks.

**Comment by NTXAggie** [comment]
> so what works for you is that you just install @fullcalendar/angular@4.4.5-beta ?
> 
> Thanks.

So far yes. I verified the drag-n-drop works, paginated calendar works, changing views works, click events works.

package.json noteable packages:
```    
    "@angular/animations": "^9.0.7",
    "@angular/cdk": "^9.1.3",
    "@angular/common": "^9.0.7",
    "@angular/compiler": "^9.0.7",
    "@angular/core": "^9.0.7",
    "@angular/forms": "^9.0.7",
    "@angular/material": "^9.1.3",
    "@angular/material-moment-adapter": "^9.1.3",
    "@angular/platform-browser": "^9.0.7",
    "@angular/platform-browser-dynamic": "^9.0.7",
    "@angular/router": "^9.0.7",
    "@fullcalendar/angular": "^4.4.5-beta",
    "@fullcalendar/core": "^4.4.0",
    "@fullcalendar/daygrid": "^4.4.0",
    "@fullcalendar/interaction": "^4.4.0",
    "@fullcalendar/timegrid": "^4.4.0",
```

**Comment by gitalvininfo** [comment]
Already done your solution but the error is gone however the calendar is not displaying. Can I ask if you have any time to kindly upload it to stackblitz so that I could be able to watch for it, if not it's okay.

Thanks btw.

**Comment by gitalvininfo** [comment]
never mind btw, already solved my bad, forgot to add FullCalendarModule in my appmodule. Thanks man, Im glad you helped me.

**Comment by pierresh** [comment]
Hello,

I get that error only when I am running unit tests (using Karma / Jasmine) on my web app, is there any way to solve this? Thanks!

**Comment by codealvarez** [comment]
> the `@fullcalendar/angular@4.4.5-beta` package has been published with all the latest changes from the angular9 branch. please let me and @irustm know if things still don't work

It works, thanks. Just type  `npm i @fullcalendar/angular@4.4.5-beta` - However is comming the 5.0.0 version soon.

**Comment by azherf** [comment]
But why should we install @fullcalendar/angular@4.4.5-beta at all ? Why can it not work without it ?
